src/tree/mod.rs

```diff
@@ -790,9 +789,16 @@ impl RatchetTree {
     pub(crate) fn path_secret(&self, index: NodeIndex) -> Option<&PathSecret> {
         // Get a Vector containing the node indices of the direct path to the
         // root from our own leaf.
-        if let Ok(dirpath) = treemath::leaf_direct_path(self.own_node_index(), self.leaf_count()) {
+        if let Ok(mut dirpath) =
+            treemath::leaf_direct_path(self.own_node_index(), self.leaf_count())
+        {
+            // Filter out blanks for which we don't have path secrets
+            let mut dirpath = dirpath
```

Comment:
No need for the `let`.

Reply:
I don't see a way to just remove it. I change `dirpath` from a vector to an iterator, so I think I have to use a `let` here. I've changed the variables to make that clearer. Let me know if I'm missing something here.